src/main/java/org/springframework/hateoas/SimpleResourceAssembler.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.hateoas;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.util.Assert;
+
+/**
+ * A {@link ResourceAssembler} based purely on the domain type, using {@code Resource<T>} as the enclosing
+ * "resource" type.
+ *
+ * @author Greg Turnquist
+ * @since 1.0
+ */
+public interface SimpleResourceAssembler<T> extends ResourceAssembler<T, Resource<T>> {
+
+	/**
+	 * Converts the given entity into a {@link Resource}.
+	 *
+	 * @param entity
+	 * @return
+	 */
+	default Resource<T> toResource(T entity) {
+		
+		Resource<T> resource = new Resource<>(entity);
+		addLinks(resource);
+		return resource;
+	}
+
+	/**
+	 * Define links to add to every individual {@link Resource}.
+	 *
+	 * @param resource
+	 */
+	default void addLinks(Resource<T> resource) {
```

Comment:
I wonder if it would be better to leave this (and the other `addLinks`) method without a default implementation. 

The current NOOP implementation doesn't seem to be the right thing to do only in very obscure cases.
So anybody implementing this interface will have to provide an implementation, but the presence of a default implementation makes it harder to see that.

Reply:
After using it, I agree. So I'll remove the default methods and tune the test cases to match.